gcp-observability/src/test/java/io/grpc/gcp/observability/logging/GcpLogSinkTest.java

```diff
@@ -98,32 +100,31 @@ public class GcpLogSinkTest {
       .putFields("rpc_id", Value.newBuilder().setStringValue(rpcId).build())
       .build();
   private Logging mockLogging;
+  private GcpLogSink spySink;
 
   @Before
   public void setUp() {
     mockLogging = mock(Logging.class);
+    spySink = spy(new GcpLogSink(destProjectName, locationTags,
+        customTags, flushLimit));
+    Mockito.doReturn(mockLogging).when(spySink).createLoggingClient();
   }
 
   @Test
   public void createSink() {
-    Sink mockSink = new GcpLogSink(mockLogging, destProjectName, locationTags,
-        customTags, flushLimit);
-    assertThat(mockSink).isInstanceOf(GcpLogSink.class);
+    assertThat(spySink).isInstanceOf(GcpLogSink.class);
```

Comment:
what does this test achieve? Or what does it really test from the "class-under-test"?